homeassistant/components/emoncms/sensor.py

```diff
@@ -139,27 +144,53 @@ async def async_setup_entry(
 ) -> None:
     """Set up the emoncms sensors."""
     config = entry.options if entry.options else entry.data
+    url = config[CONF_URL]
+    apikey = config[CONF_API_KEY]
     name = sensor_name(config[CONF_URL])
     exclude_feeds = config.get(CONF_EXCLUDE_FEEDID)
     include_only_feeds = config.get(CONF_ONLY_INCLUDE_FEEDID)
 
     if exclude_feeds is None and include_only_feeds is None:
         return
 
+    emoncms_client = EmoncmsClient(url, apikey, session=async_get_clientsession(hass))
+    emoncms_unique_id = await emoncms_client.async_get_uuid()
+    if emoncms_unique_id is None:
+        async_create_issue(
+            hass,
+            DOMAIN,
```

Comment:
Can all the checks be moved to `__init__.py`?

I think you should be using a config entry migration with `async_migrate_entry`

=> if version is 1, then check for new `unique_id` and
   => either migrate the entities and update to 2
   => or raise issue and stay on 1
=> if version is 2, then you know it's already done

Reply:
Or maybe this only works if you want to force users to migrate their database to one that supports unique ids.
If you want to continue to allow both with and without unique id, then I'm not 100% what the best solution is.

Reply:
Ok thanks done as suggested